### ISSUE TITLE
Getting all members in group chunked overwrite/ignore recursive

### ISSUE THREAD

**Issue by paddy01** [DirectoryTree/LdapRecord#409]
### Discussed in https://github.com/DirectoryTree/LdapRecord/discussions/408

<div type='discussions-op-text'>

<sup>Originally posted by **paddy01** February 24, 2022</sup>
I'm having issues with chunking results from a `LdapRecord\Models\ActiveDirectory\Group` query.
For some reason, might me working as intended, I only get the Groups not the Members of the Groups when doing

```
$ldapGroup = Group::where('sAMAccountName', '=', $groupName)->first();

$ldapGroup->members()->recursive()->chunk($queryChunk, function($objects) {
                dump($objects);
    }
});
```
this result in `$objects` only containing `LdapRecord\Models\ActiveDirectory\Group` objects.

When doing the same query but skipping chunk
```
$ldapGroup = Group::where('sAMAccountName', '=', $groupName)->first();

$orig = $ldapGroup->members()->recursive()->get();
```
ends up with `$orig` containing both `LdapRecord\Models\ActiveDirectory\Group` and `LdapRecord\Models\ActiveDirectory\User` objects ?

I'm out go get all members recursively, a.k.a. that is member of `$groupName` or any group that is member of `$groupName`.

Edit:
Actually further testing shows that `chunk` seem to completely overwrite `recursive`.
I did a test with `$ldapGroup->members()->recursive()->each` instead but ended up with the whole ldap tree instead of the selected group.</div>

**Comment by stevebauman** [comment]
Hi @paddy01, I haven't forgotten about this. `chunk()` does in-fact execute a different query that doesn't take `recursive()` into account. Going to see if I can fix this 👍

**Comment by stevebauman** [comment]
Released in [v2.16.0](https://github.com/DirectoryTree/LdapRecord/releases/tag/v2.16.0) 🎉
